llama_index/vector_stores/weaviate.py

```diff
@@ -288,11 +288,16 @@ def query(self, query: VectorStoreQuery, **kwargs: Any) -> VectorStoreQueryResul
         parsed_result = parse_get_response(query_result)
         entries = parsed_result[self.index_name]
 
-        similarities = [get_node_similarity(entry, similarity_key) for entry in entries]
-        nodes = [to_node(entry, text_key=self.text_key) for entry in entries]
+        similarities = []
+        nodes = []
+        node_idxs = []
 
-        nodes = nodes[: query.similarity_top_k]
-        node_idxs = [str(i) for i in range(len(nodes))]
+        for i, entry in enumerate(entries):
+            similarities.append(get_node_similarity(entry, similarity_key))
```

Comment:
any reason to append every similarity, but only the top-k nodes/idxs? 

I see this was the previous logic, but tbh I don't see why it does that haha

Reply:
eh, pushed a small change